KubeArmor/core/kubeUpdate.go

```diff
@@ -384,10 +384,12 @@ func (dm *KubeArmorDaemon) UpdateEndPointWithPod(action string, pod tp.K8sPod) {
 		if val, ok := dm.DefaultPostures[newEndPoint.NamespaceName]; ok {
 			newEndPoint.DefaultPosture = val
 		} else {
+			// fetch default posture from k8s configmap
+			cm := dm.GetConfigMap()
 			globalDefaultPosture := tp.DefaultPosture{
```

Comment:
The problem remains the same imo.

Consider this.

1. DefaultPosture = Block (In config Map)
2. We set dm.DefaultPosture[samplens] = Block
3. Update DefaultPosture = Audit
4. We read dm.DefaultPosture[samplens] and it already has a value Block, we would never update the value since it already exists.

Reply:
I would suggest you to create a testcase about this process and validate things with that testcase once when you make updates :)